include/execution.hpp

```diff
@@ -2461,7 +2461,110 @@ namespace _P2300::execution {
   /////////////////////////////////////////////////////////////////////////////
   // [execution.senders.adaptors.bulk]
   namespace __bulk {
+    template <class _ReceiverId, integral _Shape, class _FunId>
+      class __receiver
+        : receiver_adaptor<__receiver<_ReceiverId, _Shape, _FunId>, __t<_ReceiverId>> {
+        using _Receiver = __t<_ReceiverId>;
+        using _Fun = __t<_FunId>;
+        friend receiver_adaptor<__receiver, _Receiver>;
+
+        [[no_unique_address]] _Shape __shape_;
+        [[no_unique_address]] _Fun __f_;
+
+        template <class... _As>
+        void set_value(_As&&... __as) && noexcept 
+          requires __nothrow_invocable<_Fun, _Shape, _As...> {
+          for (_Shape __i{}; __i != __shape_; ++__i) {
+            std::invoke(__f_, __i, std::forward<_As>(__as)...);
+          }
+          execution::set_value(std::move(this->base()), std::forward<_As&&>(__as)...);
```

Comment:
`std::forward`, blech. Just a waste of compile-time resources. `static_cast<_As&&>(__as)...` is fine. `(_As&&) __as...` would also work here and be consistent with the rest of the code. (Yes, I use C-style casts for forwarding. :man_shrugging:)